db/db_impl/db_impl.cc

```diff
@@ -2085,9 +2086,26 @@ Status DBImpl::GetEntity(const ReadOptions& _read_options, const Slice& key,
   std::vector<Slice> keys;
   std::vector<ColumnFamilyHandle*> column_families;
   for (size_t i = 0; i < num_column_families; ++i) {
-    // Adding the same key slice for different CFs
-    keys.emplace_back(key);
-    column_families.emplace_back((*result)[i].column_family());
+    if (!(*result)[i].column_family()) {
+      if (s.ok()) {
+        s = Status::InvalidArgument(
+            "DB failed to query because one or more group(s) have null column "
+            "family handle");
+      }
+      (*result)[i].SetStatus(
+          Status::InvalidArgument("Column family handle cannot be null"));
+    } else if (!s.ok()) {
+      (*result)[i].SetStatus(
+          Status::Incomplete("DB not queried due to invalid argument(s) in "
+                             "another attribute group"));
+    } else {
+      // Adding the same key slice for different CFs
+      keys.emplace_back(key);
+      column_families.emplace_back((*result)[i].column_family());
+    }
+  }
```

Comment:
If we're going to fail the entire query when a null column family is encountered, it would probably make sense to go back and update the status for any earlier (valid) attribute groups as well. Then again, I think it might be simpler to just fail those specific attribute groups that don't have a column family and let the rest proceed.

Reply:
Thanks! I've decided to fix the loop to go back and update the status for earlier attribute groups.